docker/transmart-api-server/docker-entrypoint.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/sh
+set -e
+
+# Error message and exit for missing environment variable
+fatal() {
+		cat << EndOfMessage
+###############################################################################
+!!!!!!!!!! FATAL ERROR !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+###############################################################################
+			The variable with the name '$1' is unset.
+			Please specify a value in this container environment using
+			-e in docker run or the environment section in Docker Compose.
+###############################################################################
+EndOfMessage
+		exit 1
+}
+
+# Check that Postgres and Keycloak Host are configured
+# via environment variables
+[ ! -z ${PGHOST+x} ] || fatal 'PGHOST'
+[ ! -z ${PGPORT+x} ] || fatal 'PGPORT'
+[ ! -z ${KEYCLOAK_HOST+x} ] || fatal 'KEYCLOAK_HOST'
+[ ! -z ${KEYCLOAK_PORT+x} ] || fatal 'KEYCLOAK_PORT'
+
+# Fixed values, not configurable by user
+PGDATABASE=transmart
+BIOMART_USER='biomart_user'
+BIOMART_PASSWORD="${BIOMART_USER}"
+TRANSMART_API_SERVER_CONFIG_FILE="${TRANSMART_USER_HOME}/transmart-api-server.config.yml"
+
+cat > "${TRANSMART_API_SERVER_CONFIG_FILE}" <<EndOfMessage
```

Comment:
The content of this config file could be significantly reduced. Most of the properties are set by default based on the internal application.yml file that is packaged inside the application jar. 
The properties specified here will override the ones from internal (inside-jar) config, so the file generated here could be limited to something like this:

````
---
dataSource:
    driverClassName: org.postgresql.Driver
    dialect: org.hibernate.dialect.PostgreSQLDialect
    url: jdbc:postgresql://${PGHOST}:${PGPORT}/${PGDATABASE}
    username: ${PGUSER}
    password: ${PGPASSWORD}
    dbCreate: none
    pooled: true
    jmxExport: true
    logSql: false
    formatql: false
    properties:
        jmxEnabled: true
        initialSize: 5
        maxActive: 50
        minIdle: 5
        maxIdle: 25
        maxWait: 10000
        maxAge: 600000
        timeBetweenEvictionRunsMillis: 5000
        minEvictableIdleTimeMillis: 60000
        validationQuery: select 1
        validationQueryTimeout: 3
        validationInterval: 15000
        testOnBorrow: true
        testWhileIdle: true
        testOnReturn: false
        jdbcInterceptors: ConnectionState
        defaultTransactionIsolation: 2 # TRANSACTION_READ_COMMITTED

keycloak:
    realm: ${KEYCLOAK_REALM}
    bearer-only: true
    auth-server-url: ${KEYCLOAK_URL}
    resource: ${KEYCLOAK_CLIENT}
    use-resource-role-mappings: true

keycloakOffline:
    offlineToken: <offline_token>

org.transmartproject.patientCountThreshold: 5

```